kotlinx-coroutines-core/jvm/src/internal/ThreadSafeHeap.kt

```diff
@@ -20,8 +19,7 @@ public interface ThreadSafeHeapNode {
  * Synchronized binary heap.
  * @suppress **This an internal API and should not be used from general code.**
  */
-@InternalCoroutinesApi
-public class ThreadSafeHeap<T> : SynchronizedObject() where T: ThreadSafeHeapNode, T: Comparable<T> {
+internal class ThreadSafeHeap<T> : SynchronizedObject() where T: ThreadSafeHeapNode, T: Comparable<T> {
```

Comment:
Using `ThreadSafeHeap` was my explicit ask to make sure it behaves consistently when tasks are submitted from other threads. It goes not mean it will actually work, but at least will not produce some spectacularly strange behaviors. We'll need to use our own class any way when we start to commonized Test APIs.